src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -55,27 +56,40 @@ def __init__(
         self._migration_status_refresher = migration_status_refresher
         self._seen_tables: dict[str, str] = {}
         self._principal_grants = principal_grants
+        self._hiveserde_in_place_migrate = hiveserde_in_place_migrate
 
     def index(self):
         # TODO: remove this method
         return self._migration_status_refresher.index()
 
-    def migrate_tables(self, what: What, acl_strategy: list[AclMigrationWhat] | None = None):
+    def migrate_tables(
+        self, what: What, acl_strategy: list[AclMigrationWhat] | None = None, mounts_crawler: Mounts | None = None
+    ):
         if what in [What.DB_DATASET, What.UNKNOWN]:
             logger.error(f"Can't migrate tables with type {what.name}")
             return None
         all_grants_to_migrate = None if acl_strategy is None else self._gc.snapshot()
         all_migrated_groups = None if acl_strategy is None else self._group.snapshot()
         all_principal_grants = None if acl_strategy is None else self._principal_grants.get_interactive_cluster_grants()
         self._init_seen_tables()
+        # mounts will be used to replace the mnt based table location in the DDL for hiveserde table in-place migration
+        mounts = None
+        if mounts_crawler:
+            mounts = mounts_crawler.snapshot()
         if what == What.VIEW:
             return self._migrate_views(acl_strategy, all_grants_to_migrate, all_migrated_groups, all_principal_grants)
         return self._migrate_tables(
-            what, acl_strategy, all_grants_to_migrate, all_migrated_groups, all_principal_grants
+            what, acl_strategy, all_grants_to_migrate, all_migrated_groups, all_principal_grants, mounts
         )
 
     def _migrate_tables(
-        self, what: What, acl_strategy, all_grants_to_migrate, all_migrated_groups, all_principal_grants
+        self,
+        what: What,
+        acl_strategy,
+        all_grants_to_migrate,
+        all_migrated_groups,
+        all_principal_grants,
+        mounts: Iterable[Mount] | None = None,
```

Comment:
```suggestion
        mounts: list[Mount],
```

1. hidden bug: **it is not possible to consume the same iterator twice**, so it's not the right abstract type here. wrap it with `list` at the call site
2. `list[Mount]` is always better than `None | list[Mount]`.
3. replace all `Iterable[Mount]` with `list[Mount]`

Reply:
fixed